lib/bolt/util.rb

```diff
@@ -3,27 +3,23 @@
 module Bolt
   module Util
     class << self
-      def read_config_file(path, default_paths = nil, file_name = 'file')
+      # If file exists it is parsed and validated to be a valid yaml hash or empty. In the case it
+      # is empty an empty hash is returned. If the file does not exist: when'required' is true, a
+      # validation error is raised, when `required` is false the function returns `nil`.
+      def read_yaml_hash(path, required, file_name)
```

Comment:
It would be clearer if this were two methods so that the expected behavior will be clear at the call site without having to know what the second boolean argument means. That could be as simple as:

```rb
def read_optional_yaml_hash(path, file_name)
  FIle.exist?(path) ? read_yaml_hash(path, file_name) : {}
end
```

Reply:
Ah yeah that is much cleaner. Updated.